### PR TITLE
Tests: Use assert.equal and assert.notEqual more

### PR DESCRIPTION
**Pull request by AnnikaCodes**

### CODE REVIEW

File: test/sim/moves/curse.js [hunk 1]
```diff
@@ -132,7 +132,10 @@ describe('XY/ORAS Curse targetting when becoming Ghost the same turn', function
 		} else {
 			// Cartridge glitch
 			assert.notEqual(cursePartner.hp, cursePartner.maxhp); // Partner afflicted by Curse
-			assert(foeHP[0] === p2active[0].maxhp && foeHP[1] === p2active[1].maxhp); // Foes unaffected by Curse
+
+			// Foes unaffected by Curse
+			assert.equal(foeHP[0], p2active[0].maxhp);
```
Comment (Slayer95) [inline test/sim/moves/curse.js:137]:
```js
assert.fullHP
```

File: test/sim/abilities/flashfire.js [hunk 2]
```diff
@@ -75,6 +75,6 @@ describe('Flash Fire [Gen 3-4]', function () {
 		const flashFireMon = battle.p1.active[0];
 		flashFireMon.setStatus('frz');
 		battle.makeChoices();
-		assert(flashFireMon.hp !== flashFireMon.maxhp);
+		assert.notEqual(flashFireMon.hp, flashFireMon.maxhp);
```
Comment (DaWoblefet) [inline test/sim/abilities/flashfire.js:78]:
```suggestion
		assert.false.fullHP(flashFireMon.hp, flashFireMon.maxhp);
```

**Review by DaWoblefet — APPROVED** [review]

File: test/sim/moves/curse.js [hunk 3]
```diff
@@ -132,7 +132,10 @@ describe('XY/ORAS Curse targetting when becoming Ghost the same turn', function
 		} else {
 			// Cartridge glitch
 			assert.notEqual(cursePartner.hp, cursePartner.maxhp); // Partner afflicted by Curse
```
Comment (Zarel) [inline test/sim/moves/curse.js:134]:
```suggestion
			assert.false.fullHP(cursePartner); // Partner afflicted by Curse
```

**Comment by Zarel** [comment]
Personally I find the `===` syntax more readable (it's also terser) but I don't particularly care enough to argue about it.

One thing we could do is add a compile step to our tests and then compile the `===` syntax to the `.equals` syntax... but I don't want to slow down the test process. Honestly, if we ever did start building unit tests, though, I would want to do that.

**Comment by AnnikaCodes** [comment]
I like `assert.equals` because it automatically shows the expected and received values when a test fails, not for readability reasons.

**Comment by Zarel** [comment]
I know, that's why compiling `===` to the `.equals` syntax is the best of both worlds.

**Pull request merged**